gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py

```diff
@@ -770,93 +798,46 @@ def _fetch_template_and_params(self, context, update=False):
-            policy_rule_ids.extend(rule_set.get("policy_rules"))
-
-        policy_rules = context.gbp_plugin.get_policy_rules(
-            context._plugin_context, filters={'id': policy_rule_ids})
-
-        i = 0
-        fw_rule_list = []
-        for policy_rule in policy_rules:
-            policy_action_ids = policy_rule.get("policy_actions")
-            policy_actions_detail = context.gbp_plugin.get_policy_actions(
-                    context._plugin_context, filters={'id': policy_action_ids})
-            for policy_action in policy_actions_detail:
-                if policy_action["action_type"] == constants.GP_ACTION_ALLOW:
-                    classifier = context.gbp_plugin.get_policy_classifier(
-                            context._plugin_context,
-                            policy_rule.get("policy_classifier_id"))
-
-                    rule_name = "Rule_%s" % i
-                    stack_template[resources_key][rule_name] = (
-                                            self._generate_firewall_rule(
-                                                is_template_aws_version,
-                                                classifier.get("protocol"),
-                                                classifier.get("port_range"),
-                                                provider_cidr, consumer_cidr))
-
-                    fw_rule_list.append({'get_resource': rule_name})
-                    i += 1
-
-        if consumer_cidr != '0.0.0.0/0' or not fw_rule_key:
-            resource_name = 'OS::Neutron::FirewallPolicy'
-            fw_policy_key = self._get_heat_resource_key(
-                            stack_template[resources_key],
-                            is_template_aws_version,
-                            resource_name)
-            self._modify_fw_resources_name(context, stack_template,
-                                           provider_ptg,
-                                           is_template_aws_version)
-
-            stack_template[resources_key][fw_policy_key][properties_key][
-                'firewall_rules'] = fw_rule_list
+        fw_rule_keys = self._get_all_heat_resource_keys(
+            stack_template[resources_key],
+            is_template_aws_version,
+            'OS::Neutron::FirewallRule')
+        for fw_rule_key in fw_rule_keys:
+            fw_rule_resource = stack_template[resources_key][fw_rule_key][
+                properties_key]
+            if fw_rule_resource.get('destination_ip_address') == 'PTG':
+                fw_rule_resource['destination_ip_address'] = provider_cidr
+            if fw_rule_resource.get('source_ip_address') == 'PTG':
+                fw_rule_resource['source_ip_address'] = consumer_cidr
+
+    # Updates CIDR when "PTG" is specified as source or destination in Firewall
+    # rule. Firewall rules are not derived from PRS. This is different from
+    # what Cisco plans to have. For us, the Firewall rules only come from Node
```

Comment:
Suggest remove the Cisco comment. To clarify:
For Phase 2, we will continue to have firewall rules derived from the ndoe config. The only difference being that instead of an implict Allow All security Group, Cisco will support user explicitlyy PRS to allow all